### ISSUE TITLE
Support and enable strict mode

### ISSUE THREAD

**Issue by ohadschn** [mmistakes/minimal-mistakes#1225]
I introduced the fix for https://github.com/mmistakes/minimal-mistakes/issues/1223 (https://github.com/mmistakes/minimal-mistakes/commit/0a9ad5d24395e7985d99265747b511ce2b3eee28) and enabled liquid strict mode again. This time I was met with the following error:

`Liquid Exception: Liquid syntax error (line 21): Expected end_of_string but found pipe in "name in __names | sort" in /var/lib/gems/2.3.0/gems/minimal-mistakes-jekyll-4.5.1/_includes/group-by-array, included in _pages/category-archive.html`

Perhaps you could enable `liquid: error_mode: strict` in the demo site's `_config.yml` so that once fixed, these don't slip in?

## Environment information

- **Minimal Mistakes version:** 4.5.1
- **Jekyll version:** 3.5.2
- **GitHub Pages hosted:** no (hosted locally)
- **Operating system:** Bash on Ubuntu on Windows (WSL)

**Comment by mmistakes** [comment]
I fixed this already in https://github.com/mmistakes/minimal-mistakes/commit/c4ca06f4b9c96bf6d1f12996c91e0103f0630e56 for the `group-by-array` include in the root and under `/docs`. Perhaps you didn't pull down the latest `/docs/_includes/group-by-array`?

Testing the demo site and `/test` sites with `error_mode: strict` come out error free now.

**Comment by ohadschn** [comment]
My bad, thank you for clarifying.
Could you release a new gem version so I can update my CI with `error_mode: strict`?

**Comment by mmistakes** [comment]
Not ready to push a release just yet, but you don't need me to do that.

You can just pull from what's on master by replacing `gem "minimal-mistakes-jekyll"` in your `Gemfile` with the following:

```ruby
gem 'minimal-mistakes-jekyll', :git => 'https://github.com/mmistakes/minimal-mistakes.git'
```

And if you need something more granular you can pull from specific commits, branches, tags, etc. Consult [Bundler's documentation](http://bundler.io/v1.15/guides/git.html) for the particulars.

**Comment by ohadschn** [comment]
Fair point, I guess I could just take the current `master` tip (https://github.com/mmistakes/minimal-mistakes/commit/2080469952cbde9325dfbb721f5b33b0612d243d).

May I ask what is holding you from releasing a new version, and whether you have some estimation on when a new version would be released? Perhaps I'll wait too :)

**Comment by mmistakes** [comment]
Just life, my day job, family commitments, etc etc etc 😉

**Comment by ohadschn** [comment]
Ah yes, those pesky things :)

Let me rephrase - do you see any reason not to take the current `master` tip (https://github.com/mmistakes/minimal-mistakes/commit/2080469952cbde9325dfbb721f5b33b0612d243d)?
